lib/ranked-model/ranker.rb

```diff
@@ -170,12 +179,26 @@ def update_index_from_position
         end
       end
 
+      def possibly_rebalance_and_position(min, max)
```

Comment:
I think this could be improved in terms of how it's used and reads :) I'll have a think.